purchase_amendment/model/purchase_order_amendment.py

```diff
@@ -0,0 +1,278 @@
+            title = _('Reason for amending')
+            message += "<h3>%s</h3><p>%s</p>" % (title, self.reason)
+        return message
+
+    @api.multi
+    def do_amendment(self):
+        self.ensure_one()
+        purchase = self.purchase_id
+        purchase.message_post(body=self._message_content())
+        self.item_ids.split_lines()
+        return True
+
+    @api.multi
+    def wizard_view(self):
+        view = self.env.ref('purchase_amendment.view_purchase_order_amendment')
+
+        return {
+            'name': _('Enter the amendment details'),
+            'type': 'ir.actions.act_window',
+            'view_type': 'form',
+            'view_mode': 'form',
+            'res_model': 'purchase.order.amendment',
+            'views': [(view.id, 'form')],
+            'view_id': view.id,
+            'target': 'new',
+            'res_id': self.id,
+            'context': self.env.context,
+        }
+
+
+class PurchaseOrderAmendmentItem(models.TransientModel):
+    _name = 'purchase.order.amendment.item'
+
+    amendment_id = fields.Many2one(comodel_name='purchase.order.amendment',
+                                   string='Amendment',
+                                   required=True,
+                                   ondelete='cascade',
+                                   readonly=True)
+    purchase_line_id = fields.Many2one(comodel_name='purchase.order.line',
+                                       string='Line',
+                                       required=True,
+                                       readonly=True)
+    procurement_id = fields.Many2one(comodel_name='procurement.order',
+                                     readonly=True)
+    procurement_group_id = fields.Many2one(comodel_name='procurement.group',
+                                           related='procurement_id.group_id',
+                                           readonly=True)
+    origin = fields.Char(readonly=True)
+    move_id = fields.Many2one(comodel_name='stock.move',
+                              readonly=True)
+    original_qty = fields.Float(string='Ordered',
+                                digits_compute=dp.get_precision('Product UoS'),
+                                readonly=True)
+    new_qty = fields.Float(string='Amend',
+                           digits_compute=dp.get_precision('Product UoS'))
+    state = fields.Char(readonly=True)
+    product_id = fields.Many2one(related='purchase_line_id.product_id',
+                                 readonly=True)
+    product_uom_id = fields.Many2one(related='purchase_line_id.product_uom',
+                                     readonly=True)
+
+    def affected_sale_lines(self):
+        move_chain = self.env['stock.move']
+        current_move = self.move_id
+        if current_move:
+            move_chain |= current_move
+            current_move = current_move.move_dest_id
+
+        return move_chain.mapped('procurement_id.sale_line_id')
```

Comment:
If you don't have the sale module installed this is resulting in an error: KeyError: 'sale_line_id'.
IMHO the sale part of the code should be moved to a separate module. And add a hook to the split_lines method that can be used in this separate module.